project/Build.scala

```diff
@@ -517,6 +517,35 @@ object Build {
       pomIncludeRepository := { _ => false }
   )
 
+  /** Constants for the `verScheme` parameter of `publishSettings`.
+   *
+   *  sbt does not define constants in its API for `versionScheme`. It
+   *  specifies some strings instead. We use the following version schemes,
+   *  depending on the artifacts and the versioning policy in `VERSIONING.md`:
+   *
+   *  - `"strict"` for artifacts whose public API can break in patch releases (e.g., `compiler`)
+   *  - `"pvp"` for artifacts whose public API can break in minor releases (e.g., `linker`)
+   *  - `"semver-spec"` for artifacts whose public API can only break in major releases (e.g., `library`)
+   *    or in severe releases (e.g., `testAdapter`).
+   *
+   *  The version scheme of Severe changes may need to be revisited if and when
+   *  we publish such a release, as its impact on the version numbers is
+   *  decided on a case-by-case basis, as explained in `VERSIONING.md`.
+   *
+   *  See also https://www.scala-sbt.org/1.x/docs/Publishing.html#Version+scheme
+   */
+  object VersionScheme {
+    final val BreakOnPatch = "strict"
+    final val BreakOnMinor = "pvp"
+    final val BreakOnSevere = "semver-spec"
```

Comment:
What happens if two artifacts that differ only in version have different values for `VersionScheme`?

Reply:
According to https://github.com/sbt/librarymanagement/pull/339, the version scheme of the “winner” artifact (which is always the one with the highest version number?) applies. In short, changing the version scheme of an artifact during its life is not supported.

Reply:
https://github.com/sbt/sbt/discussions/6572#discussioncomment-947655 confirmed the behavior. However, I disagree that the use case is not supported. To me that is exactly the behavior we want: if we change the version scheme later for something stricter, the resolution algorithm will correctly pick the stricter value of `versionScheme` and report the correct incompatibilities.

Reply:
What if we have:
- 1.0.0 semver
- 1.1.0 pvp
- 1.2.0 semver

Will this accurately detect a conflict between 1.0.0 / 1.2.0? (Or can we simply never make versioningScheme less strict anymore?)

Reply:
> Or can we simply never make versioningScheme less strict anymore?

I think that's the answer. Also it makes sense to me, independently of the specifics of the resolution algo: you can't become more lenient as time progresses, since you'll still have old releases that were breaking on minor. This is not different than having pvp since v1.0.0, and then changing to semver in v1.2.0. You don't need the double-change to create a problem; just going once in the direction from stricter to more lenient is the problem.

Reply:
I'm more and more getting the impression, that our definition / plan for severe changes is broken. IMO if that happens, we might be better off decoupling SJS linker and and binary IR version.

Reply:
> if we change the version scheme later for something stricter, the resolution algorithm will correctly pick the stricter value of `versionScheme` and report the correct incompatibilities.

AFAIU, it works by chance in this situation, but what if you change the scheme later for something less strict?

Reply:
As I mentioned above: becoming *less* strict in the future is not valid (for fundamental reasons, not just because of the specific resolution algorithm). But becoming *stricter* in the future is allowed.